sumologic/sumologic_gcp_source.go

```diff
@@ -17,17 +17,17 @@ type GCPThirdPartyRef struct {
 }
 
 type GCPResource struct {
-	ServiceType    string            `json:"serviceType"`
-	Authentication GCPAuthentication `json:"authentication,omitempty"`
-	Path           GCPPath           `json:"path,omitempty"`
+	ServiceType    string             `json:"serviceType"`
+	Authentication *GCPAuthentication `json:"authentication"`
```

Comment:
I feel as we are using struct pointers, we should make these fields `omitempty` rather than changing the `GCPAuthentication` and `GCPPath` structs.